extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/restclient/OpenTelemetryClientFilter.java

```diff
@@ -72,20 +72,25 @@ public OpenTelemetryClientFilter(final OpenTelemetry openTelemetry) {
 
     @Override
     public void filter(final ClientRequestContext request) {
-        if (isReactiveClient(request)) {
-            return;
-        }
-
         Context parentContext = Context.current();
         if (instrumenter.shouldStart(parentContext, request)) {
             Context spanContext = instrumenter.start(parentContext, request);
-            Scope scope = spanContext.makeCurrent();
+            Scope scope = QuarkusContextStorage.INSTANCE.attach(getVertxContext(request), spanContext);
             request.setProperty(REST_CLIENT_OTEL_SPAN_CLIENT_CONTEXT, spanContext);
             request.setProperty(REST_CLIENT_OTEL_SPAN_CLIENT_PARENT_CONTEXT, parentContext);
             request.setProperty(REST_CLIENT_OTEL_SPAN_CLIENT_SCOPE, scope);
         }
     }
 
+    private static io.vertx.core.Context getVertxContext(final ClientRequestContext request) {
+        io.vertx.core.Context vertxContext = (io.vertx.core.Context) request.getProperty(VERTX_CONTEXT_PROPERTY);
+        if (vertxContext == null) {
+            return QuarkusContextStorage.getVertxContext();
```

Comment:
I see `QuarkusContextStorage.getVertxContext()` can return null. 
I don't fully understand when, is it only if a user spawns their own thread and calls the client from it?
Do we still have tracing in this case?

Reply:
We only use the Vert.x Context when we have a Vert.x duplicated Context available. If not, we fall back to the ThreadLocal Context Storage (for instance when you are tracing a CDI Bean outside Vert.x)

Reply:
cool